### PR TITLE
ISPN-11958 Infinispan Archetypes Getting Started Docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11958

### CODE REVIEW

File: archetypes/pom.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <version>11.0.1-SNAPSHOT</version>
 
     <name>Infinispan Archetypes</name>
-    <description>Archetypes for the Infinispan project</description>
+    <description>Maven Archetypes for Infinispan projects.</description>
```
Comment (pruivo) [inline archetypes/pom.xml:18]:
```suggestion
    <description>Infinispan Maven Archetypes</description>
```
wdyt?

File: documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
+[id='mvn_archetypes']
+= Creating {brandname} Projects
+{brandname} provides a set of Maven archetypes that you can use to generate
+skeleton projects with sample code and `pom.xml` files. You can then import
+these projects into your IDE and start building and running applications with
+{brandname} capabilities.
+
+.Prerequisites
+
+Before you can use the {brandname} archetypes, you need to do the following:
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc:10]: this is incorrect. 
you don't need to compile the Infinispan project. The archetypes will be available in maven central.

File: documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+. Change to the cloned repository and compile {brandname}.
++
+----
+$ cd infinispan
+$ mvn clean install -DskipTests
+----
+
+.Procedure
+
+. Navigate to the `archetypes` directory in the {brandname} repository.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc:28]: not required. you can generate the code from any folder you want.

File: documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
+[id='mvn_archetypes']
+= Creating {brandname} Projects
+{brandname} provides a set of Maven archetypes that you can use to generate
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc:3]: would it make sense to add a link to https://maven.apache.org/guides/introduction/introduction-to-archetypes.html? for users that aren't aware of what it is.

File: documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+* `embedded` Creates a sample application that embeds {brandname}.
+* `store` Creates a sample implementation of a custom cache store.
++
+<2> Specifies the {brandname} version to use, for example:
++
+`-DarchetypeVersion={ispn_version}`
++
+. When prompted, define _artifactId_, _groupId_, and _version_ values for your project.
+. Import your new {brandname} project into your IDE.
+. Run your project application from the directory where the `pom.xml` file resides as follows:
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc:54]: put a note saying the "store" archetype doesn't create any running code. (i.e. no Application.java)

File: archetypes/README.md [hunk 6]
```diff
@@ -1,19 +1,23 @@
-# The Infinispan Archetypes project
+# Infinispan Archetypes
```
Comment (ryanemerson) [inline archetypes/README.md:1]: Now that we have proper documentation, we can probably remove this file entirely. I included it it in the initial PR mainly as a primer for yourself.

File: documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc (hunk 4 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_using_mvn_archetypes.adoc:3]: yeah, that'd be good. obviously I could do with reading that too...

File: archetypes/README.md (hunk 6 above)
Comment (oraNod) [reply archetypes/README.md:1]: maybe it's good to have something there. I've included a link to the getting started guide. perhaps that is all that's needed.

**Comment by pruivo** [comment]
integrated! thanks @oraNod !

**Pull request merged**